### PR TITLE
Complete loggers by new level and message handler for user warnings

### PR DESCRIPTION
**Pull request by ssalatsk**
*CLOUDBLD-4640

Signed-off-by: Serhii Salatskyi <ssalatsk@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [n/a] JSON/YAML configuration changes are updated in the relevant schema
- [n/a] Changes to metadata also update the documentation for the metadata
- [n/a] Pull request has a link to an osbs-docs PR for user documentation updates
- [n/a] New feature can be disabled from a configuration file

### CODE REVIEW

**Comment by ssalatsk** [comment]
This PR is blocked until [osbs-client PR](https://github.com/containerbuildsystem/osbs-client/pull/1015) is merged

File: atomic_reactor/plugins/pre_resolve_composes.py [hunk 1]
```diff
@@ -131,9 +131,11 @@ def allow_inheritance(self):
         self.log.info("Inheritance requested in config file")
 
         if is_scratch_build(self.workflow) or is_isolated_build(self.workflow):
-            self.log.warning(
-                "Inheritance is not allowed for scratch or isolated builds. "
-                "Skipping inheritance.")
+            warning = {
+                "message": ("Inheritance is not allowed for scratch or isolated builds. "
+                            "Skipping inheritance.")
```
Comment (tkdchen) [inline atomic_reactor/plugins/pre_resolve_composes.py:136]: In this context, the `()` is unnecessary.

**Review by tkdchen — COMMENTED** [review]
LGTM

File: atomic_reactor/plugins/pre_resolve_composes.py [hunk 2]
```diff
@@ -131,9 +131,9 @@ def allow_inheritance(self):
         self.log.info("Inheritance requested in config file")
 
         if is_scratch_build(self.workflow) or is_isolated_build(self.workflow):
-            self.log.warning(
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_resolve_composes.py:134]: I'd like to keep also original log on the server side as a warning

File: atomic_reactor/__init__.py [hunk 3]
```diff
@@ -17,12 +17,8 @@
 
 from atomic_reactor.version import __version__  # noqa
 
-try:
-    from osbs.constants import ATOMIC_REACTOR_LOGGING_FMT
-except ImportError:
-    # not available in osbs < 0.41
-    fmt = '%(asctime)s platform:%(arch)s - %(name)s - %(levelname)s - %(message)s'
-    ATOMIC_REACTOR_LOGGING_FMT = fmt
+from osbs.constants import (ATOMIC_REACTOR_LOGGING_FMT, USER_WARNING_LEVEL)
```
Comment (MartinBasti) [inline atomic_reactor/__init__.py:20]: Why parentheses here?

File: atomic_reactor/plugins/pre_resolve_composes.py (hunk 2 above)
Comment (rcerven) [reply atomic_reactor/plugins/pre_resolve_composes.py:134]: I am actually wondering IF logging user warnings shouldn't log also regular warning

Comment (ssalatsk) [reply atomic_reactor/plugins/pre_resolve_composes.py:134]: Good point :+1:
To close it, I opened a [PR](https://github.com/containerbuildsystem/osbs-client/pull/1017) in osbs-client to send warnings automatically.

Comment (MartinBasti) [reply atomic_reactor/plugins/pre_resolve_composes.py:134]: I'm not sure if we want this

Comment (rcerven) [reply atomic_reactor/plugins/pre_resolve_composes.py:134]: yeah just create another log as warning as well,

we might want to log different message in user & warn logs

Comment (ssalatsk) [reply atomic_reactor/plugins/pre_resolve_composes.py:134]: Fixed

**Review by rcerven — APPROVED** [review]

**Review by MartinBasti — APPROVED** [review]

**Comment by MartinBasti** [comment]
Please provide release notes

**Comment by ssalatsk** [comment]
Release notes:
atomic-reactor can store user warnings in their logs

**Pull request merged**